src/etherscan/verifyCollectionPartyContract.ts

```diff
@@ -0,0 +1,250 @@
+ */
+export const verifyCollectionParty = async (
+  collectionPartyAddress: string,
+  collectionPartyCreationTx: string
+): Promise<{ isSuccess: boolean; guid?: string }> => {
+  const contractCreationBytecode = await getContractCreationBytecode(
+    collectionPartyCreationTx
+  );
+
+  // NOTE: this must match the logic variable of the deployed CollectionPartyFactory contract https://github.com/PartyDAO/partybid/blob/main/contracts/CollectionPartyFactory.sol#L32
+  const logicAddress = partyLogicAddresses.collection;
+  const logicAddressHexEncoded = `000000000000000000000000${logicAddress.replace(
+    "0x",
+    ""
+  )}`;
+  const constructorArgumentsBytecode = getConstructorArgumentsBytecode(
+    contractCreationBytecode,
+    logicAddressHexEncoded
+  );
+
+  try {
+    // NOTE: these arguments must match the hardhat configuration from: https://github.com/PartyDAO/partybid/blob/main/hardhat.config.js
+    const params = getParamsForEtherscan({
+      apiKey: config.etherscan.apiKey,
+      module: "contract",
+      action: "verifysourcecode",
+      sourceCode: NonReceivableInitializedProxySourceCode,
+      contractAddress: collectionPartyAddress,
+      codeFormat: "solidity-single-file",
+      contractName: "NonReceivableInitializedProxy",
+      compilerVersion: "v0.8.9+commit.e5eed63a",
+      optimizationUsed: 1,
+      runs: 999999,
+      constructorArguments: constructorArgumentsBytecode,
+      licenseType: 3, // enum for MIT
+    });
+    const resp = await axios.post(config.etherscan.apiBase, params);
+
+    if (!(resp && resp.data)) {
+      throw new Error(
+        `Invalid response from Etherscan API for ${collectionPartyAddress}`
+      );
+    }
+    const { data } = resp;
+    console.log("etherscan:: data", data);
+    if (data.status === "1" && data.result) {
+      const guid = data.result;
+      console.log(
+        `verifyCollectionParty:: successfully verified collection party contract for ${collectionPartyAddress}. guid ${guid}.`
+      );
+      return { isSuccess: true, guid };
+    } else {
+      throw new Error(
+        `Could not verify collection party contract for ${collectionPartyAddress}`
+      );
+    }
+  } catch (error) {
+    console.error(error);
+    return { isSuccess: false };
+  }
+  return { isSuccess: false };
+};
+
+/**
+ * verify proxy contract on etherscan if the event is a start event
+ * @param {PartyEvent} event
+ * @returns true/false if verified
+ */
+export const verifyCollectionPartyContractForEvent = async (
+  event: PartyEvent
+): Promise<boolean> => {
+  if (event.party.partyType !== "collection") {
+    return;
+  }
+
+  const { partyAddress } = event.party;
+  const existingGuid = await getEtherscanVerifyContractGuid(partyAddress);
+  if (existingGuid) {
+    // we have already made an api request to etherscan to verify this contract
+    return;
+  }
+
+  // todo: check etherscan API to see if collection party source code is verified
```

Comment:
do we still need to add this? `getEtherscanVerifyProxyContractGuid`

Reply:
not for this file! we use that in the `verifyProxyContract` file